homeassistant/components/jvc_projector/sensor.py

```diff
@@ -0,0 +1,76 @@
+"""Sensor platform for JVC Projector integration."""
+
+from __future__ import annotations
+
+from jvcprojector import const
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import JvcProjectorDataUpdateCoordinator
+from .const import DOMAIN
+from .entity import JvcProjectorEntity
+
+JVC_SENSORS = (
+    SensorEntityDescription(
+        key="power",
+        translation_key="jvc_power_status",
+        device_class=SensorDeviceClass.ENUM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        options=[
+            const.STANDBY,
+            const.ON,
+            const.WARMING,
+            const.COOLING,
+            const.ERROR,
+        ],
+    ),
+    SensorEntityDescription(
+        key="input",
+        translation_key="jvc_input",
+        device_class=SensorDeviceClass.ENUM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        options=[
+            const.HDMI1,
+            const.HDMI2,
+            const.NOSIGNAL,
+        ],
```

Comment:
As the input is changeable, this should not be a sensor but a select entity.
Please remove it from this PR and create a follow-up, where you add it as a select entity

Reply:
Good callout. Looks like Select Entity's have a `current_option` property which could read the state from the data coordinator.

Reply:
Sorry is not correct. This one is read only, should not be a Select Entity.
The actions to change input are already present in Remote.
This is not a entity to change items so it should remain as is

Reply:
Please explain why this is incorrect and should remain as it is.

> This one is read only, should not be a Select Entity.
> The actions to change input are already present in Remote.

If you can change it, then it is not read-only.
Thinking about user experience it would be better to be a select entity so the user can change it easily instead of having two different entities: one to get the value and another to set it.

Reply:
sorry I do not understand what you mean with "If you can change it, then it is not read-only.".
I cannot change it. The sensor can not change it.
What in the code gives you the idea that I can change it?
This reads the status from the projector.

Not sure I follow you. if you can help me to understand better
when you have a sensors it's always "changeable" as the sensor has multiple states (2 or more). The point is who changes the state... and is not the sensor, the sensor reads and reports here the status.

Reply:
I don't speak about the representation in HA but on the projector itself.

The projector's input can be changed via the HA remote entity. Therefore, the input (on the projector) is not read-only.
Adding a sensor entity in HA for it has the downside that we can only read the information and not easily set it.
The select entity provides an easy way to read and set a value.
From the user experience perspective, it is more comfortable to set the input with one entity instead of having two, one for getting it and one for setting it.

Reply:
Another way to think about this. The remote integration really only exists to capture functionality not inherently exposed by native HA functionality/entities (for instance the numeric keys on a remote, or the channel up/down button). In this case functionality exists in HA to allow you to change projector inputs via the Select Entity, which can also report the state of the entity. Its the same for the power functionality of the projector, which is exposed in the base Entity in the integration.

Reply:
SteveEasley In reality is not 100% correct in terms of the projector :) 
The projector has 2 different set of commands one is to change the input, the power etc.. the other set of commands is to read the status.
The remote component is focused on the "set" commands, the sensor works on the "read" commands.
I can understand the HA view but I think we must start from the device itself.
If you have a projector you know you have an "info" command that reports the status and a "hdmi 1" that changes the status, it's natural to assume the same behavior in HA for a projector device. Otherwise the behavior in HA is not coherent with the behavior of the projector itself.

Specifically from the JVC documentation:
Binary Code: 21 - Operational command
Binary Code: 3F - Information command

The sensor will only use 3F codes :)